src/codegen/mod.rs

```diff
@@ -760,20 +770,40 @@ impl CodeGenerator for Type {
                     !is_opaque &&
                     inner_item.expect_type().canonical_type(ctx).is_enum()
                 {
-                    tokens.append_all(quote! {
-                        pub use
-                    });
-                    let path = top_level_path(ctx, item);
-                    tokens.append_separated(path, quote!(::));
-                    tokens.append_all(quote! {
-                        :: #inner_rust_type  as #rust_name ;
+                    let mut path = quote! {};
+                    path.append_separated(
+                        top_level_path(ctx, item),
+                        quote!(::),
+                    );
+
+                    tokens.append_all(match alias_style {
+                        AliasVariation::TypeAlias => quote! {
+                            pub use #path :: #inner_rust_type as #rust_name ;
+                        },
+                        AliasVariation::NewType | AliasVariation::NewTypeDeref => quote! {
+                            pub struct #rust_name (pub #path :: #inner_rust_type) ;
```

Comment:
Woops. I misread what you said and made `Deref` (that won't work) for enums.

What should be the behavior in case of `AliasVariation::NewTypeDeref`? Just a `warn!` or should it panic?

Reply:
I don't understand the last commit either.

The condition that leads to this code right now is:

```rust
                if inner_rust_type.to_string().chars().all(|c| match c {
                    // These are the only characters allowed in simple
                    // paths, eg `good::dogs::Bront`.
                    'A'..='Z' | 'a'..='z' | '0'..='9' | ':' | '_' | ' ' => true,
                    _ => false,
                }) && outer_params.is_empty() &&
                    !is_opaque &&
                    inner_item.expect_type().canonical_type(ctx).is_enum()
                {
```

Why can't it just be:

```rust
                if inner_rust_type.to_string().chars().all(|c| match c {
                    // These are the only characters allowed in simple
                    // paths, eg `good::dogs::Bront`.
                    'A'..='Z' | 'a'..='z' | '0'..='9' | ':' | '_' | ' ' => true,
                    _ => false,
                }) && outer_params.is_empty() &&
                    !is_opaque &&
                    alias_style == AliasVariation::TypeAlias &&
                    inner_item.expect_type().canonical_type(ctx).is_enum()
                {
```

Then the other stuff would just work, and it'd be simpler, wouldn't it?

Reply:
Simpler and less error prone, last 2 commits weren't even using `#[repr(transparent)]` on that part. Sorry, thought you meant something else.